src/System.Diagnostics.Process/tests/ProcessTests.netcoreapp.cs

```diff
@@ -123,5 +129,264 @@ public void BothArgumentSetAndArgumentListSet()
             testProcess.StartInfo = psi;
             Assert.Throws<InvalidOperationException>(() => testProcess.Start());
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_ThrowsPlatformNotSupportedException()
+        {
+            var process = new Process();
+            Assert.Throws<PlatformNotSupportedException>(() => process.Kill(entireProcessTree: true));
+            Assert.Throws<PlatformNotSupportedException>(() => process.Kill(entireProcessTree: false));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_True_ProcessNotStarted_ThrowsInvalidOperationException()
+        {
+            var process = new Process();
+            Assert.Throws<InvalidOperationException>(() => process.Kill(entireProcessTree: true));
+        }
+
+        
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)] // Currently, remote processes are only supported on Windows. If that changes, adjust accordingly.
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_True_CalledByNonLocalProcess_ThrowsInvalidOperationException()
+        {
+            Process currentProcess = Process.GetCurrentProcess();
+            Process process;
+
+            try
+            {
+                process = Process.GetProcessById(currentProcess.Id, "127.0.0.1");
+            }
+            catch (InvalidOperationException)
+            {
+                // As we can't detect reliably if performance counters are enabled, 
+                // we silently abort on InvalidOperationExceptions since this test
+                // can only run if the attempt to get the process succeeded.
+                return;
+            }
+
+            Assert.Throws<NotSupportedException>(() => process.Kill(entireProcessTree: true));
+
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_True_CalledOnCallingProcess_ThrowsInvalidOperationException()
+        {
+            var process = Process.GetCurrentProcess();
+            Assert.Throws<InvalidOperationException>(() => process.Kill(entireProcessTree: true));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public void Kill_EntireProcessTree_True_CalledOnTreeContainingCallingProcess_ThrowsInvalidOperationException()
+        {
+            Process containingProcess = CreateProcess(() =>
+            {
+                Process parentProcess = CreateProcess(() => RunProcessAttemptingToKillEntireTreeOnParent());
+
+                parentProcess.Start();
+                parentProcess.WaitForExit();
+
+                return parentProcess.ExitCode;
+
```

Comment:
ditto

Reply:
same comment for other places

Reply:
Thanks! Fixed in several places! (Sorry about that!)